networking/Field_Control_Networking_Spec.md

```diff
@@ -0,0 +1,83 @@
+# Field Control Networking Specificiation
+
+## Contents
+
+1. Overview
+2. Setting up the Clients and Network
+3. Connecting to the Field Control Network
+4. Establishing Communication Links
+5. Sending Messages
+
+## 1. Overview
+
+A robot interacts with two networks:  the Student network and the Field Control (FC) network.  The former is the network over which students communicate with their robot via provided mini-routers.  The latter--the focus of this spec--is the network over which students' robots interact with FC over the course of a match.
+
+Clients (field control, driver stations, and robots) are assigned static IPs and communicate over TCP connections.
+
+FC sends out messages to initiate robot-to-driver-station pairing and robot mode changes.  FC listens to heartbeat messages from robots.
+
+## 2. Setting Up the Clients and Network
+
+### Clients
+
+The FC network will have up to eight clients, as follows:
+
+- One FC client.  This is the FC computer running the match.  A PiE computer+monitor. 
+- Four driver station clients.  This acts as the student-robot interface (via Dawn and Runtime), and as the robot-field control interface (see "Establishing Communication Links").  A wifi-enabled laptop or  computer+monitor.
+- Up to four robot clients.  A wifi-enabled Beagle Bone Black (via wifi dongle).  One per student robot.
+
+### Assigning IP Addresses
+
+Every client will have a statically assigned IP address.  This is feasible due to the small number of total clients (1 per robot ( less than 30 in total), 4 driver stations, and one field control center).
+
+Consider the following example static IP scheme ("P" = subnet prefix):
+
+- FCl:  P.255
+- Driver Stations:  P.254-251
+- Robots:  P.TeamNumber
+
+## 3. Connecting to the Field Control Network
+
+### Configuration
+
+A robot can connect to at most one network at a time, which must be either the Student network or the FC network.  The Student network is to be listed as higher priority than the FC Network.  This can be set each robot's network config file.
```

Comment:
An example of one reason we'd need to deploy a custom config: say we need to update the settings for field control connections due to last-minute discovering issues with our field router config.